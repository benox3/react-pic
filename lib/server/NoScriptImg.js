import React, { Component } from 'react';
import convertReactToHTMLStyle from '../utils/convertReactToHTMLStyle';

export default class NoScriptImg extends Component {
  render() {
    const { alt, style, image} = this.props;

    return (
      <noscript dangerouslySetInnerHTML={{
        __html:
          '<img ' +
            `alt='${alt}' ` +
            `style='${convertReactToHTMLStyle(style)}' ` +
            `src='${image.url}' />`
      }} />
    );
  }
}
