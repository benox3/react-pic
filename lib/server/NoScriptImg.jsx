import React, { PropTypes } from 'react';
import convertReactToHTMLStyle from '../utils/convertReactToHTMLStyle';

const NoScriptImg = ({ alt, style, image }) => (
  <noscript
    dangerouslySetInnerHTML={{
      __html:
        '<img ' +
          `alt='${alt}' ` +
          `style='${convertReactToHTMLStyle(style)}' ` +
          `src='${image.url}' />`,
    }}
  />
);

NoScriptImg.propTypes = {
  alt: PropTypes.string.isRequired,
  style: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  ).isRequired,
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
  }).isRequired,
};

export default NoScriptImg;
