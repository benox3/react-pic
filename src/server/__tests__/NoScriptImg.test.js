import React from 'react';

import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import NoScriptImg from '../NoScriptImg';

describe('NoScriptImg', () => {
  const props = {
    alt: 'heart',
    image: {
      width: 290,
      url: 'http://placehold.it/290?text=â™¥',
    },
    style: {
      margin: 0
    }
  };

  describe('rendering', () => {
    it('should render', () => {
      expect(mount(<NoScriptImg {...props} />).html()).to.not.equal(null);
    });

    it('should render a noscript tag', () => {
      expect(
        mount(<NoScriptImg {...props} />).find('noscript').exists()
      ).to.be.true;
    });
  });

  describe('snapshots', () => {
    it('should match base snapshot', () => {
      expect(renderer.create(<NoScriptImg {...props} />)).toMatchSnapshot();
    });
  });
});