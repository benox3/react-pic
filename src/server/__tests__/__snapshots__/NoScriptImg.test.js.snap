// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoScriptImg snapshots should match base snapshot 1`] = `
<noscript
  dangerouslySetInnerHTML={
    Object {
      "__html": "<img alt='heart' style='margin:0;' src='http://placehold.it/290?text=â™¥' />",
    }
  }
/>
`;
